tla-import/src/main/scala/at/forsyte/apalache/io/json/TlaToJson.scala

```diff
@@ -53,146 +54,187 @@ import at.forsyte.apalache.tla.lir.io.TypeTagPrinter
  * @param factory A json factory for the `T` variant of JsonRepresentation
  * @tparam T Any class extending JsonRepresentation
  */
-class TlaToJson[T <: JsonRepresentation](factory: JsonFactory[T])(implicit typTagPrinter: TypeTagPrinter)
+class TlaToJson[T <: JsonRepresentation](
+    factory: JsonFactory[T], locatorOpt: Option[SourceLocator] = None
+)(implicit typeTagPrinter: TypeTagPrinter)
     extends JsonEncoder[T] {
-  val kindFieldName: String = "kind"
-  val typeFieldName: String = "type"
+  import TlaToJson._
 
   implicit def liftString: String => T = factory.fromStr
```

Comment:
Is there a good reason for using implicit conversions here? AFAIK, this may stop working in Scala 3. In general, it's hard to refactor code that is using implicits, as we have seen recently.

Reply:
Yes, being able to write `s -> "abc"` instead of `s -> factory.fromStr("abc")` in like 20 spots. This is far from the kind of implicits used in (un)typed predefs, since it's declared in the same spot as it is used, so it's more of a convenience shorthand.